src/main/java/com/streamforge/realm/twitch/oauth/camel/TwitchAuthRoute.java

```diff
@@ -0,0 +1,96 @@
+package com.streamforge.realm.twitch.oauth.camel;
+
+import com.streamforge.realm.twitch.oauth.dtos.TwitchUserTokenDto;
+import com.streamforge.realm.twitch.oauth.transformer.TwitchAuthDataTransformer;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static com.streamforge.realm.Constants.*;
+import static org.apache.camel.builder.PredicateBuilder.and;
+
+@Component
+public class TwitchAuthRoute extends RouteBuilder {
+    @Value("${streamforge.twitch.token-uri}")
+    private String tokenURI;
+
+    private TwitchHttpQueryBuilder twitchHttpQueryBuilder;
+
+    @Autowired
+    public TwitchAuthRoute(TwitchHttpQueryBuilder twitchHttpQueryBuilder) {
+        this.twitchHttpQueryBuilder = twitchHttpQueryBuilder;
+    }
+
+    private final Consumer<Exchange> HTTP_QUERY_PARAMS_TO_MAP = exchange -> {
+        Message in = exchange.getIn();
+        String queryParams = (String) in.getHeader(Exchange.HTTP_QUERY);
+        List<NameValuePair> mappedParams = new ArrayList<>();
+        try {
+            URIBuilder uriBuilder = new URIBuilder(Q_MARK + queryParams);
+            mappedParams = uriBuilder.getQueryParams();
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
```

Comment:
Could you log this error instead of using prinStackTrace, please

Reply:
Sure, thats a lot of work to do